bin/i18n/sync/metrics.rb

```diff
@@ -0,0 +1,62 @@
+require 'benchmark'
+require 'cdo/aws/metrics'
+require 'aws-sdk-ec2'
+require 'net/http'
+require 'socket'
+
+module I18n
+  module Sync
+    module Metrics
+      I18N_METRICS_NAMESPACE = 'I18n'.freeze
+
+      def self.runtime(process)
+        result = nil
+        runtime = Benchmark.realtime {result = yield}
+        log_metric(
+          :RuntimeTest,
+          runtime,
+          [
+            {name: "SyncStep", value: "in"},
+            {name: "MethodName", value: process}
+          ],
+          'Seconds'
+        )
+
+        result
+      end
+
+      def self.machine_id
+        @machine_id ||= begin
+          metadata_endpoint = 'http://169.254.169.254/latest/meta-data/'
+          Net::HTTP.get(URI.parse(metadata_endpoint + 'instance-id'))
+        rescue
+          'local_machine'
+        end
+      end
+
+      # logging metrics into cloudwatch under i18n name space
+      # @param metric_name [Symbol] A metric name
+      # @param metric_value [Object] A metric value
+      # @param addtl_dimensions [Array<Hash>] additional dimensions
+      # @option opts [String] Metric units. Valid Values see CloudWatch MetricDatum:
+      # https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html
+      def self.log_metric(metric_name, metric_value, addtl_dimensions, metric_units='None')
```

Comment:
do you expect to use the metric_units param?

Reply:
yes, by default if is not provided, it sets the metrics to none. We want to be able to provide different units depending on the metric.

Reply:
ok cool, I might update the comment to say something like "e.g. seconds, milliseconds, etc" so that it's more clear without going to the link for the reader